symFileManager.py

```diff
@@ -220,3 +230,13 @@ def MaybeEvict(self, freeEntriesNeeded):
     #print "MRU after: " + str(self.sMruSymbols)
     LogDebug("Cache occupancy after MaybeEvict: " + str(self.sCacheCount) + "/" + str(maxCacheSize))
 
+  def _FetchSymbolsFromPaths(self, paths, pathSuffix, joinFunction, fetchFunction, cachePathSuffix):
```

Comment:
So, we have basically 4 symbol sources: memory cache, disk cache, local paths and remote paths. What I think would make the code easier to maintain is a "Fetcher" abstraction. I am going to work on something.